### PR TITLE
fix(ZNTA-959,ZNTA-1000): Add try catch for copy version task

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-959
https://zanata.atlassian.net/browse/ZNTA-1000

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java [hunk 1]
```diff
@@ -114,50 +114,57 @@ public void copyVersion(@Nonnull String projectSlug,
...
-        projectIterationDAO.flush();
-
-        // clear any cache that has been loaded in this new version before copy
-        // completed
-        versionStateCacheImpl.clearVersionStatsCache(newVersion.getId());
-        log.info("copy version end: copy {} to {}, {}", projectSlug
-                + ":" + versionSlug, projectSlug + ":" + newVersionSlug,
+        } catch (Exception e) {
+            log.error(
+                "Error during copy version from project '{}' {} to {}: {}",
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java]: If you leave out the final `{}`, slf4j should log the exception stack trace with the log entry.

File: zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java [hunk 2]
```diff
@@ -114,50 +114,57 @@ public void copyVersion(@Nonnull String projectSlug,
...
+                projectSlug, versionSlug, newVersionSlug, e);
+            e.printStackTrace();
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java]: One (extra) log for this exception should be enough. We don't need `e.printStackTrace()` too.

File: zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java [hunk 3]
```diff
@@ -114,50 +114,56 @@ public void copyVersion(@Nonnull String projectSlug,
...
-
-        // clear any cache that has been loaded in this new version before copy
-        // completed
-        versionStateCacheImpl.clearVersionStatsCache(newVersion.getId());
-        log.info("copy version end: copy {} to {}, {}", projectSlug
-                + ":" + versionSlug, projectSlug + ":" + newVersionSlug,
+        } catch (Exception e) {
+            log.error(
+                "Error during copy version from project '{}' {} to {}.",
+                projectSlug, versionSlug, newVersionSlug);
```
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/service/impl/CopyVersionServiceImpl.java]: What happened to `e`? You still have to pass the exception as a parameter.

**Comment by seanf** [comment]
@aeng This branch has conflicts.

**Comment by seanf** [comment]
:+1:

**Comment by djansen-redhat** [comment]
:white_check_mark: Verified log entry appears (fix later)

**Comment by seanf** [comment]
Anything useful in the log entry?

On 7 April 2016 at 15:52, Damian Jansen <notifications@github.com> wrote:

> [image: :white_check_mark:] Verified log entry appears (fix later)
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1127#issuecomment-206707228>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

**Pull request merged**